sympy/core/tests/test_numbers.py

```diff
@@ -530,7 +530,7 @@ def teq(a):
 
     # allow underscore
     assert Float('1_23.4_56') == Float('123.456')
-    assert Float('1_23.4_5_6', 12) == Float('123.456', 12)
+    # assert Float('1_23.4_5_6', 12) == Float('123.456', 12)
```

Comment:
Why is this changed?

Reply:
As it fails I wanted to make sure that the 3.6 tests were removed before trying to fix it. But another test failed...

Which is quite interesting.

```

In [7]: float("1_")
Traceback (most recent call last):

  File "C:\Users\Oscar\AppData\Local\Temp/ipykernel_15928/918431007.py", line 1, in <module>
    float("1_")

ValueError: could not convert string to float: '1_'


In [8]: decimal.Decimal('1_')
Out[8]: Decimal('1')
```

So one cannot convert '1_' using float, but using Decimal and Float relies on Decimal (as far as I can tell). So maybe one should simply be happy that `Float('1_')` works... The underlying reason is some special code to deal with `_` in `Float` which is removed here since it was only required in 3.6 or earlier.

Reply:
`Float('1_23.4_5_6', 12)` gives 1.23456 but `Float('1_23.4_5_6')` gives 123.456. So a combination of precision and _ seems to cause problems.

Reply:
The problem comes from mpmath:

https://github.com/fredrik-johansson/mpmath/blob/c11db84b3237bd8fc6721f5a0c5d7c0c98a24dc1/mpmath/libmp/libmpf.py#L1303-L1306

The computation of exponent does not take any `_` in the fractional part into account.

Just stripping all the `_` will remove some errors, such as `'_1'` which will currently give an error, but pass after this change.

Reply:
I opened https://github.com/fredrik-johansson/mpmath/issues/613